src/Eshopworld.WorkerProcess/CosmosDistributedLock/CosmosDistributedLockStore.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using Eshopworld.Core;
+using EShopworld.WorkerProcess.Configuration;
+using EShopworld.WorkerProcess.Telemetry;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Options;
+using Polly;
+using Polly.Retry;
+
+namespace EShopworld.WorkerProcess.CosmosDistributedLock
+{
+    public class CosmosDistributedLockStore : ICosmosDistributedLockStore
+    {
+        private readonly IDocumentClient _documentClient;
+        private readonly IOptions<CosmosDataStoreOptions> _options;
+        private readonly AsyncRetryPolicy _retryPolicy;
+        private readonly IBigBrother _telemetry;
+
+        public CosmosDistributedLockStore(IDocumentClient documentClient, IOptions<CosmosDataStoreOptions> options, IBigBrother telemetry)
+        {
+            _documentClient = documentClient;
+            _options = options;
+            _telemetry = telemetry;
+            _retryPolicy = CreateRetryPolicy();
+        }
+
+        public async Task InitialiseAsync()
+        {
+            await _documentClient.CreateDatabaseIfNotExistsAsync(new Database { Id = _options.Value.Database })
+                .ConfigureAwait(false);
+
+            await CreateLockClaimsCollection();
+        }
+
+        private async Task CreateLockClaimsCollection()
+        {
+            var collectionDefinition = new DocumentCollection
+            {
+                Id = _options.Value.DistributedLocksCollection
+            };
+
+            await _documentClient.CreateDocumentCollectionIfNotExistsAsync(
```

Comment:
What's the partition key that you use for your lock collections?

Reply:
PartitionKey = /id
I use id the lock name so a partition key is not really needed.